pyaedt/edb_core/edb_data/hfss_simulation_setup_data.py

```diff
@@ -0,0 +1,1700 @@
+
+    @property
+    def name(self):
+        """Get name of this sweep."""
+        return self._edb_sweep_data.Name
+
+    @name.setter
+    def name(self, value):
+        """Set name of this sweep"""
+        self._edb_sweep_data.Name = value
+        self._update_sweep()
+
+    @property
+    def sweep_type(self):
+        """Get sweep type."""
+        return
+
+    @property
+    def frequencies(self):
+        """Get list of frequencies points."""
+        return list(self._edb_sweep_data.Frequencies)
+
+    @property
+    def adaptive_sampling(self):
+        """Use adaptive sampling.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.AdaptiveSampling
+
+    @property
+    def adv_dc_extrapolation(self):
+        """Turn on advanced DC Extrapolation.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.AdvDCExtrapolation
+
+    @property
+    def auto_s_mat_only_solve(self):
+        """Auto/Manual SMatrix only solve.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.AutoSMatOnlySolve
+
+    @property
+    def enforce_causality(self):
+        """Enforce causality during interpolating sweep.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.EnforceCausality
+
+    @property
+    def enforce_dc_and_causality(self):
+        """Enforce DC point and causality.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.EnforceDCAndCausality
+
+    @property
+    def enforce_passivity(self):
+        """Enforce passivity during interpolating sweep.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.EnforcePassivity
+
+    @property
+    def freq_sweep_type(self):
+        """Sweep type (interpolation, discrete or Broadband Fast).
+
+        Returns
+        -------
+        str
+        """
+        return self._edb_sweep_data.FreqSweepType.ToString()
+
+    @property
+    def interp_use_full_basis(self):
+        """Use Full basis.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.InterpUseFullBasis
+
+    @property
+    def interp_use_port_impedance(self):
+        """Use Port impedance.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.InterpUsePortImpedance
+
+    @property
+    def interp_use_prop_const(self):
+        """Use propagation constants.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.InterpUsePropConst
+
+    @property
+    def interp_use_s_matrix(self):
+        """Use S matrix.
+
+        Returns
+        -------
+        bool
+        """
+        return self._edb_sweep_data.InterpUseSMatrix
+
+    @property
+    def max_solutions(self):
+        """Max solutions.
```

Comment:
Is "Max" short here for "Maximum"? We want to avoid abbreviating words to avoid loss in clarity.